src/plugins/routing.js

```diff
@@ -27,6 +27,14 @@ function localePathFactory (i18nPath, routerPath) {
       route = { name: route }
     }
 
+    // if route has a path defined but no route name, generate path without route
+    if (route.path && !route.name) {
+      // prepend locale if current is not default and not using different domains
+      const appendPath = !(locale === defaultLocale && strategy === STRATEGIES.PREFIX_EXCEPT_DEFAULT) && !this[i18nPath].differentDomains
+      const fullPath = (appendPath ? `/${locale}${route.path}` : route.path)
+      return fullPath
```

Comment:
Not going through `router.resolve` means that this code won't handle `params` or `query`.

(I don't know if router will be able to handle route without name though...)

Reply:
I think the router should be able to resolve the it without the name. [Example here](https://router.vuejs.org/guide/essentials/navigation.html#programmatic-navigation) uses path + query in one of samples, and I'm assuming that's resolved through the router the same way the others samples are. Will fix.